utils/prof_utils.py

```diff
@@ -616,6 +616,126 @@ def est_sn_from_prof(prof_data, period, alpha=3.):
 
     return [sn, sn_e, scattered]
 
+#---------------------------------------------------------------
+def analyse_pulse_prof(prof_data, period, verbose=True):
+    """
+    Estimates the signal to noise ratio from a pulse profile. Returns are in list form. Will return more for verbose==True setting, explained below.
+    NOTE: user must supply EITHER a betprof path OR prof_data and period of the pulse profile.
+    Based on code oringally writted by Nick Swainston.
+    This is the old version of 'est_sn_from_prof' but is useful when we can't fit gaussians
+
+    Parameters:
+    -----------
+    prof_data: list
+        A list of floats that contains the pulse profile. Default
+    period: float
+        The pulsar's period in ms
+    verbose: boolean
+        OPTIONAL - Determines whether to return more detailed information. Detailed below. Default: True
+    Returns:
+    --------
+    sn: float
+        The estimated signal to noise ratio
+    u_sn: float
+        The estimated signal to noise ratio's its uncertainty
+    flags: list
+        VERBOSE - a list of flagged data points
+    w_equiv_bins: float
+        VERBOSE - the equivalent width of the profile measured in bins
+    u_w_equiv_bins: float
+        VERBOSE - the uncertaintiy in w_equiv_bins
+    w_equiv_ms: float
+        VERBOSE - the equivalent width of the profile measured in ms
+    u_w_equiv_ms: float
+        VERBOSE - the uncertainty in w_equiv_ms
+    scattering: float
+        VERBOSE: - the scattering width in
+    u_scattering: float
+    scattered: boolean
+        VERBOSE - when true, the profile is highly scattered
+    """
+
+    nbins = len(prof_data)
+    #centre the profile around the max
+    shift = -int(np.argmax(prof_data))+int(nbins)//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find sigma and check if profile is scattered
+    sigma, flags = submit_to_database.sigmaClip(prof_data, alpha=3., tol=0.01, ntrials=100)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [undefined name 'submit_to_database'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4921142)